Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -337,8 +338,15 @@ namespace ROS2::Utils
             const bool needsVisual = visuals.contains(unresolvedUrfFileName);
             const bool needsCollider = colliders.contains(unresolvedUrfFileName);
 
-            AZ::IO::Path targetPathAssetDst(importDirectoryDst / resolvedPath.Filename());
-            AZ::IO::Path targetPathAssetTmp(importDirectoryTmp / resolvedPath.Filename());
+            AZStd::string filename = resolvedPath.Filename().String();
+            auto count = countFilenames[filename]++;
+            if (count > 0)
+            {
+                filename = AZStd::string::format("duplicate_%u_%s", count, filename.c_str());
```

Comment:
Suggestion: maybe use "%s_dup_%u" as a slightly less cumbersome name, and so that it will easier to find in an alphabetical sort when browsing through files. For example, in the duplicate "link2.stl" case, you'd have "link2" and "link2_dup_1", so they both show up when looking for link2* instead of needing to search for *link2*.

Reply:
oh, whoops, that's what i _meant_ to suggest, i didn't think about the fact that the %s was including the extension too. sorry about that, thanks for the change!